iOS Example Frame/iOS Example Frame/ViewController/HomeViewController.swift

```diff
@@ -62,47 +63,40 @@ class HomeViewController: UIViewController {
 
   @IBAction private func showOtherTheme(_ sender: Any) {
     customizeNavigationBarAppearance(backgroundColor: UIColor(red: 23 / 255, green: 32 / 255, blue: 30 / 255, alpha: 1),
-                                       foregroundColor: .green)
+                                     foregroundColor: .green)
     let viewController = Factory.getOtherPaymentViewController { [weak self] result in
       self?.handleTokenResponse(with: result)
     }
     navigationController?.pushViewController(viewController, animated: true)
   }
 
   @IBAction private func getApplePayData(_ sender: Any) {
-    // Use example Apple Pay payment data.
-    guard let paymentDataURL = Bundle.main.url( forResource: "example_apple_pay_payment_data", withExtension: "json") else {
-      print("Unable to get URL of Apple Pay payment data.")
+    let paymentNetworks: [PKPaymentNetwork] = [.amex, .discover, .masterCard, .visa]
+    let paymentItem = PKPaymentSummaryItem(label: "Test item", amount: NSDecimalNumber(value: 12.99))
+    guard PKPaymentAuthorizationViewController.canMakePayments(usingNetworks: paymentNetworks) else {
+      showAlert(with: "Unable to make Apple Pay transaction.")
       return
     }
-
-    let paymentData: Data
-
-    do {
-      paymentData = try Data(contentsOf: paymentDataURL)
-    } catch {
-      print(error.localizedDescription)
+    let request = PKPaymentRequest()
+    request.currencyCode = "USD"
+    request.countryCode = "US"
+    request.merchantIdentifier = "com.checkout.merchant.io.test.ios"
+    request.merchantCapabilities = PKMerchantCapability.capability3DS
+    request.supportedNetworks = paymentNetworks
+    request.paymentSummaryItems = [paymentItem]
+    guard let paymentVC = PKPaymentAuthorizationViewController(paymentRequest: request) else {
+      showAlert(with: "Unable to present Apple Pay authorization.")
       return
     }
-
-    // Potential Task: public struct ApplePay in Checkout SDK needs a public init otherwise will be treated as internal
-    let applePay = ApplePay(tokenData: paymentData)
-
-    checkoutAPIService.createToken(.applePay(applePay)) { status in
-      switch status {
-        case .failure(let error):
-          self.showAlert(with: error.localizedDescription)
-        case .success(let tokenDetails):
-          self.showAlert(with: tokenDetails.token)
-      }
-    }
+    paymentVC.delegate = self
+    present(paymentVC, animated: true, completion: nil)
```

Comment:
If we do want to add this, we could keep the HomeVC cleaner by moving this logic to a separate file. In the project structure, next to ThemeDemo, we could add `PaymentRequestCreator.swift` with a static function that enables someone to try at will

```swift
enum PaymentRequestCreator {
    static func createMockApplePayToken(delegate: PKPaymentAuthorizationViewControllerDelegate) -> PKPaymentAuthorizationViewController? {
        // Perform all the customisation we just added and return VC if `canMakePayments` passed
    }
}
```

Here, we'd be looking to have simpler logic as in:
```swift
@IBAction private func getApplePayData(_ sender: Any) {
    guard let paymentAuthorizationVC = PaymentRequestCreator. createMockApplePayToken(delegate: self) else {
        // Retrieve sample token from file and pass to `processToken`
        let paymentData = getSampleDataFromJSONFile()
        processToken(paymentData)
        return
    }
    present(paymentVC, animated: true, completion: nil)
}

private func processToken(_ paymentData: Data) {
    // This is the original code
    let applePay = ApplePay(tokenData: paymentData)
    checkoutAPIService.createToken(.applePay(applePay)) { status in
        switch status {
        case .failure(let error):
            self.showAlert(with: error.localizedDescription)
        case .success(let tokenDetails):
            self.showAlert(with: tokenDetails.token)
       }
     }
}
```